### PR TITLE
Add button click callback to PopOver

### PR DESCRIPTION
**Pull request by alvichan**
**Issue #:** 

**Description of changes:**
Add button click callback to PopOver.

**Testing**
1. Have you successfully run `npm run build:release` locally?
Yes

2. How did you test these changes?
Yes. Ran release script locally and manual test

3. If you made changes to the component library, have you provided corresponding documentation changes?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added the callback fired when the render button is clicked to `PopOver` UI component
```
Comment (devalevenkatesh) [inline CHANGELOG.md:18]: Nit: Could you please add a "." at the end of the sentence here.

Comment (devalevenkatesh) [inline CHANGELOG.md:18]: Just a thought, should we reword this log? For example:
Added an optional prop `onRenderButtonClick` to pass a callback function to the `PopOver` UI component. This callback will be called when the `PopOver` UI component is clicked.

**Review by devalevenkatesh — CHANGES_REQUESTED** [review]

File: src/components/ui/PopOver/index.tsx [hunk 2]
```diff
@@ -40,6 +40,8 @@ export interface PopOverProps
   renderButton?: (isActive: boolean) => {};
   /** Alternative to renderButton, defines the function to render the full popover button element (as opposed to just its contents). This is used if you want full control over the button rendering. The button must forwardRef */
   renderButtonWrapper?: (isActive: boolean, props: any) => {};
+  /** The callback fired when the render button is clicked. */
```
Comment (devalevenkatesh) [inline src/components/ui/PopOver/index.tsx:43]: Is this on the render button click or the PopOver element click?

Comment (alvichan) [reply src/components/ui/PopOver/index.tsx:43]: It is the "^" button click.

File: src/components/ui/PopOver/index.tsx [hunk 3]
```diff
@@ -40,6 +40,8 @@ export interface PopOverProps
...
+  onButtonClick?: (isOpen: boolean) => void;
```
[the remark this answers is not in the document]
Comment (devalevenkatesh) [reply src/components/ui/PopOver/index.tsx:44]: Sorry I thought this is the render button, if it is the PopOver element click itself then, we can just call this as `onClick` right? I checked we have one more prop named `closeOnClick` hence, thought of suggesting so.

**Review by devalevenkatesh — APPROVED** [review]

**Pull request closed without merging**